src/joystick/windows/SDL_dinputjoystick.c

```diff
@@ -1105,6 +1126,10 @@ SDL_DINPUT_JoystickUpdate(SDL_Joystick * joystick)
 {
     HRESULT result;
 
+    if (dinput == NULL) {
```

Comment:
This shouldn't be called if dinput is NULL. If that's not the case, then that's a bug in the higher level code and should be fixed.